src/pheval/analyse/benchmark_generator.py

```diff
@@ -0,0 +1,69 @@
+from collections import defaultdict
+from dataclasses import dataclass
+from typing import Callable
+
+from pheval.analyse.benchmarking_data import BenchmarkRun
+from pheval.analyse.disease_prioritisation_analysis import benchmark_disease_prioritisation
+from pheval.analyse.gene_prioritisation_analysis import benchmark_gene_prioritisation
+from pheval.analyse.rank_stats_writer import RankStatsWriter
+from pheval.analyse.run_data_parser import TrackInputOutputDirectories
+from pheval.analyse.variant_prioritisation_analysis import benchmark_variant_prioritisation
+from pheval.constants import (
+    DISEASE_PLOT_FILE_PREFIX,
+    DISEASE_PLOT_Y_LABEL,
+    GENE_PLOT_FILE_PREFIX,
+    GENE_PLOT_Y_LABEL,
+    VARIANT_PLOT_FILE_PREFIX,
+    VARIANT_PLOT_Y_LABEL,
+)
+
+
+@dataclass
+class BenchmarkRunOutputGenerator:
+    """Base class for recording data required for generating benchmarking outputs."""
+
+    prioritisation_type_file_prefix: str
+    y_label: str
+    benchmark_function: Callable[
```

Comment:
Rename `benchmark_function` as `generate_benchmark_run`? Isn't the result of the `Callable` a `BenchmarkRun`? Although perhaps `BenchmarkRun` should be called `BenchmarkRunResults` which fits the class documentation `Analysis results for a run`, in which case the method would be better called  `generate_benchmark_run_results` as this is what is does.